### ISSUE TITLE
Hybrid Heater modes not available

### ISSUE THREAD

**Issue by thumbnut** [rstrouse/nodejs-poolController-dashPanel#61]
I have the IntelliCenter i10PS and the UltraTemp ETi connected via RS-485. The heat modes are not available on the dashPanel. The radio buttons don't do anything. 
The drop-down selections are functional in the IntelliCenter2 app, so I know the heater is being properly recognized by the i10PS.

![image](https://user-images.githubusercontent.com/29102016/185764800-bb81d141-c119-4277-94b0-d14e69d22a26.png)

![image](https://user-images.githubusercontent.com/29102016/185764889-b35c4aa1-0468-4c2e-b02b-2162147395f7.png)

**Comment by rstrouse** [comment]
Finally somebody with IntelliCenter and a hybrid heater.  Can I get you to select  the heat modes 1 by 1 from the IntelliCenter app while doing a replay capture.  Please do them in order starting with off and working up to Hybrid-dual mode.

**Comment by thumbnut** [comment]
[replay.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/9388105/replay.zip)
Here you go.
For reference, I first set the mode to Hybrid Dual Mode before starting the capture. Then I selected in the order you see in the above screen shot. I selected the Verbose level of information in the options. If you need more (or less) let me know.

**Comment by rstrouse** [comment]
Pull njsPC.  I added the heat modes and sources.  However, I would like you to supply another replay.  This time check the `Capture Configuration Reload` checkbox and let it finish so that the upper right status indicator returns to Ready before pressing the `End Capture` message.

**Comment by thumbnut** [comment]
I pulled the latest and rebooted.  Modes are now displayed - Perfect!
![image](https://user-images.githubusercontent.com/29102016/185818154-49ffc5e4-4877-467e-b5c5-7fbc4b691371.png)

Even though the modes are shown I still had to use the Pentair interface to switch modes. While the capture dialog is open the njsPC dashboard is inactive.
Also this time I also turned the heater on and then off at the end of the sequence. When the heater was on the njsPC still showed the heater off.
![image](https://user-images.githubusercontent.com/29102016/185818401-56d1f78c-dd05-45f0-97df-61b5ce106c20.png)

The file is quite a bit larger this time.

[replay2.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/9390341/replay2.zip)

**Comment by rstrouse** [comment]
Thanks for that.  We are reading out the configuration data perfectly.  However, I am a bit confused since this replay contains items that were previously not known.  I need the exact order in which you make the changes so please change the heat mode in the following order using the IntelliCenter2 web application.

You can simply close the dialog for the capture then come back to it later to end the capture.  However, I need you to make these changes on the IntelliCenter2 web application and not dashPanel.

* Off
* Hybrid - Gas Only Mode
* Hybrid - Heat Pump Only Mode
* Hybrid - Hybrid Mode
* Hybrid - Dual Mode

I just need to be sure that this is the order of the changes since all I see are numbers.

**Comment by thumbnut** [comment]
Done exactly as requested. I did this exact sequence the last time and I did it using the IntelliCenter2 web app, but I added the step of turning the heater on and then off at the end of the sequence. The reason I did that is because the dashPanel always shows the heater as OFF even when it is heating. This time I left that extra step out.
[replay3.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/9397866/replay3.zip)

**Comment by rstrouse** [comment]
Thanks I will look at this later tonight.  It looks like they took another direction with the heat modes on the Hybrid ETi for IntelliCenter.   It is much better than how it was implemented in Touch.

**Comment by rstrouse** [comment]
I have pushed an update that will fix the heat modes.  However, I don't know whether we have the heat status.  Now I would like a capture when the heater is in the following states.

* Heating with heat pump
* Heating with gas
* Heating with dual

Can I get a capture like that.  Let me know what order these are in.

**Comment by thumbnut** [comment]
Here is the capture file.  I had to start with Hybrid-Gas Only, then Hybrid-Heat Pump, then Hybrid-Hybrid (yes, this is a mode. It lets the heater decide which is more efficient), then Hybrid-Dual.
I had to do it in this order because going from heat pump to gas and back to heat pump starts the 5 minute compressor delay.
I ended up placing a remote ethernet webcam out at the OCP so I could hear the compressor kick in.

[replay4.zip](https://github.com/rstrouse/nodejs-poolController-dashPanel/files/9425965/replay4.zip)
.

**Comment by thumbnut** [comment]
Seems like I should be sending you a capture with just commanding the heater on and then off.
Currently dashPanel is not showing this state correctly.  Let me know if you need that.

![image](https://user-images.githubusercontent.com/29102016/186708113-5a66653f-d128-4989-a700-b1ac61ba52a5.png)

![image](https://user-images.githubusercontent.com/29102016/186708341-3f68bc05-2547-4788-9a9c-e5d526230566.png)

**Comment by rstrouse** [comment]
Pull njsPC and dashPanel.  Please hard refresh your browser.  Sadly it looks like the heater status reported by IntelliCenter is only whether the heater is on or off.  With Nixie I was able to determine which heater modes were currently active.  This should be showing a flame when it is on now.

**Comment by thumbnut** [comment]
Actually if you look at the screen shot above it is showing the mode correctly as dual, so it is reporting the mode. I just changed the mode to "gas only" and the njsPC obediently displayed the change.  Just missing the "On" status and flame. I can send a new capture with the heater being turned off and then on again if that will help.

![image](https://user-images.githubusercontent.com/29102016/186733777-eaf6666c-8c56-4b20-8374-f96f25d10c7f.png)

![image](https://user-images.githubusercontent.com/29102016/186733940-bd57bf2f-59b5-4b28-a9d6-9ce7533329a8.png)

**Comment by rstrouse** [comment]
In your screenshot the heater status is off.  That flame does not show up until the thermostat from IntelliCenter calls for heat.  The setpoint is 88 and the current temp is 89,

**Comment by thumbnut** [comment]
The mode control works bidirectionally. I changed the mode to Hybrid from the dashPanel and the Pentair complied accordingly. I would say that part is fixed.  Thanks to old Mr Sol the temp is already above my set point so no flame on the Pentair panel.

![image](https://user-images.githubusercontent.com/29102016/186734987-6fb80159-8363-4fe8-93cb-fbe13d8c5e77.png)

![image](https://user-images.githubusercontent.com/29102016/186735171-8ba0c391-f305-44f3-97ec-4dbad732f9c5.png)

**Comment by rstrouse** [comment]
Your heater will not engage until the Set Point is set above the current pool temp.  Try setting the Set Point to 95 to see if the flame shows up.

**Comment by thumbnut** [comment]
That did work.  Also turned back to off when I dropped the temp set point down to 88.  I would call this one closed.

Thanks again for your great work!

![image](https://user-images.githubusercontent.com/29102016/186736648-2d997258-cc63-46f7-8a86-93e0b1b59a9d.png)

![image](https://user-images.githubusercontent.com/29102016/186736689-c01fe202-b52b-484f-8940-df748614bb3a.png)

**Comment by rstrouse** [comment]
Do a CTRL+F5 on your browser window.  There should be a flame in dashPanel if you pulled the latest.

**Comment by thumbnut** [comment]
Son of a gun!

![image](https://user-images.githubusercontent.com/29102016/186751377-4fba7217-429d-45bf-9629-dca6514f983b.png)

**Comment by rstrouse** [comment]
Much better.  Let us know if you have any other issues.  I am closing this for now so open another issue if you experience any other problems.

**Comment by thumbnut** [comment]
One thing I notice is that the "Time to off" does not reflect the Pentair settings for "Economy Time"
![image](https://user-images.githubusercontent.com/29102016/186753214-3f8f5b79-5730-47e0-97c4-1526b1700bb9.png)

In the clip from my previous comment above you can see that right after I started the heater it showed 59 minutes left, rather than 4 hours and 59 minutes. Here it shows 47 minutes left after it has been shut off and it continues to count down. There are features running so I am not sure if the big pool button should still be green with the heater off.

![image](https://user-images.githubusercontent.com/29102016/186753389-f3fc7502-b4ff-4e10-a185-c9f2e4e3edd9.png)
